core/src/main/java/org/elasticsearch/index/query/HasChildQueryParser.java

```diff
@@ -133,159 +107,10 @@ public Query parse(QueryShardContext context) throws IOException, QueryParsingEx
             throw new QueryParsingException(parseContext, "[has_child] requires 'type' field");
```

Comment:
should we remove this validation on query and type from parser if we add it to validate on the builder. The idea is that we call validate right after parsing (on the coord node), that applies to java api so can be seen as a centralized validation for both rest layer and java api.